crates/bdk/src/wallet/mod.rs

```diff
@@ -95,6 +95,90 @@ pub struct Wallet<D = ()> {
     secp: SecpCtx,
 }
 
+/// A structure to update [`Wallet`].
+///
+/// It updates [`bdk_chain::keychain::KeychainTxOutIndex`], [`bdk_chain::TxGraph`] and [`local_chain::LocalChain`] atomically.
+#[derive(Debug, Clone)]
+pub struct WalletUpdate<K, A> {
```

Comment:
```suggestion
pub struct Update {
```

`K` and `A` are `KeychainKind` and `ConfirmationTimeAnchor` and are invariant at the moment.